### PR TITLE
[Transform] Allow Disabling Adding Reverse Edges for Self Loops 

### PR DESCRIPTION
**Pull request by mufeili**
## Description

@zjost 

This addresses #3669 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/transform/functional.py [hunk 1]
```diff
@@ -714,13 +714,16 @@ def to_bidirected(g, copy_ndata=False, readonly=None):
     return g
 
 def add_reverse_edges(g, readonly=None, copy_ndata=True,
-                      copy_edata=False, ignore_bipartite=False):
+                      copy_edata=False, ignore_bipartite=False, exclude_self=False):
```
Comment (zjost) [inline python/dgl/transform/functional.py:717]: If it's likely meaningless in most cases, should the default be True?

**Review by BarclayII — APPROVED** [review]

**Comment by BarclayII** [comment]
I think it's OK to set `exclude_self` to True by default.  Feel free to merge after changing it.

**Comment by mufeili** [comment]
@zjost @BarclayII Changed.

@BarclayII You will need to mention this breaking change in the release note.

**Comment by mufeili** [comment]
It seems that the CI got stuck here @VoVAllen 

/home/jenkins/agent/workspace/dgl_fix_reverse/src/graph/transform/metis_partition_hetero.cc:78: Partition a graph with 2708 nodes and 10556 edges into 3 parts and get 303 edge cuts

[08:56:26] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:27856].

[08:56:27] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:27859].

[08:56:28] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:27862].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:38687].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:52691].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:55191].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:56445].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:60809].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:47011].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:33255].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:53693].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:36957].

[08:56:29] /home/jenkins/agent/workspace/dgl_fix_reverse/src/rpc/tensorpipe/tp_communicator.cc:98: TPReceiver starts to wait on [tcp://172.31.15.233:33901].

**Comment by VoVAllen** [comment]
Noticed. However this PR should be able to merge since ci for the PR works well.

**Pull request merged**